core/src/main/scala/io/qbeast/core/transform/Transformation.scala

```diff
@@ -47,12 +47,11 @@ trait OrdinalTransformation extends Transformation {
 /**
  * Identity transformation.
  */
-case class IdentityTransformation(newVal: Any) extends Transformation {
+case class IdentityToZeroTransformation(identityValue: Any) extends Transformation {
 
   @inline
   override def transform(value: Any): Double = value match {
-    case v: Number =>
-      v.byteValue()
+    case v: Number if v == identityValue => 0.0
   }
```

Comment:
Is the `if v == identityValue` really needed?

Reply:
I'm not entirely sure, but if this transformation is trying to transform a value that is not the original one, it should, at least, be noticed (because that can lead to the bad behavior of the index).